test/e2e/test_helm_install_upgrade.go

```diff
@@ -205,8 +205,7 @@ func TestHelmUpgrade(t *testing.T) {
 				},
 			},
 		},
-		/**
-		// TODO(Jintao): This test is disabled. After a new nightly image is available which uses KIC 3.4.1, we can enable it.
+
 		{
 			name:             "upgrade from nightly to current",
 			fromVersion:      "nightly",
```

Comment:
This will require some more changes as now nightlies are being pushed to a separate repository: https://hub.docker.com/r/kong/nightly-gateway-operator-oss/tags

(for EE that's https://hub.docker.com/r/kong/nightly-gateway-operator/tags) 

So we'll need to:

- not make the repo hardcoded and allow override in EE repo
- allow overriding the repository in testcases in this test.